{
  ExceptionCallback callback=null;
synchronized (callbackLock) {
    if (this.callback != null) {
      callback=this.callback;
    }
 else {
      unhandled.add(exception);
    }
  }
  logger.info("Handling exception: " + exception.getClass().getName());
  if (callback != null) {
    logger.info("Exception caught on TestExceptionStrategy and was sent to callback.",exception);
    callback.onException(exception);
  }
 else {
    logger.info("Exception caught on TestExceptionStrategy but there was no callback set.",exception);
  }
  return event;
}
